va/va.go

```diff
@@ -466,17 +466,24 @@ func (va *ValidationAuthorityImpl) performRemoteValidation(
 		response *vapb.ValidationResult
 		err      error
 	}
+	results := make(chan *rvaResult, len(va.remoteVAs))
 
-	results := make(chan *rvaResult)
+	ctx, cancel := context.WithCancel(ctx)
+	defer cancel()
 
 	for _, i := range rand.Perm(len(va.remoteVAs)) {
 		remoteVA := va.remoteVAs[i]
 		go func(rva RemoteVA, out chan<- *rvaResult) {
 			res, err := rva.PerformValidation(ctx, req)
-			out <- &rvaResult{
+			select {
+			case out <- &rvaResult{
 				hostname: rva.Address,
 				response: res,
 				err:      err,
+			}:
+			case <-ctx.Done():
+				// Context canceled, exit the goroutine.
+				return
```

Comment:
I'd recommend removing this branch. Now that the results chan is buffered to the full length of possible outputs, writing to it (on line 479) is guaranteed not to block. Putting a `case <-ctx.Done()` in doesn't change anything.

Reply:
This was added to facilitate early termination of the goroutines, not prevent blocking on the channel.

Reply:
Right. I'm saying the goroutine will terminate at the same time even without the `case <-ctx.Done()`, because the only other thing that's happening here is the channel write, which doesn't block.

Reply:
Oh, fair enough.